tests/Tests/ORM/Tools/Pagination/LimitSubqueryOutputWalkerTest.php

```diff
@@ -46,6 +46,22 @@ private function replaceDatabasePlatform(AbstractPlatform $platform): void
         $this->entityManager->getConnection()->setDatabasePlatform($platform);
     }
 
+    public function testSubqueryClonedCompletely(): void
+    {
+        $query = $this->createQuery('SELECT p FROM Doctrine\Tests\ORM\Tools\Pagination\MyBlogPost p');
+        $query->setParameter('dummy-param', 123);
+        $query->setHint('dummy-hint', 'dummy-value');
+        $query->setCacheable(true);
+
+        $walker = new LimitSubqueryOutputWalker($query, new Query\ParserResult(), []);
+
+        self::assertTrue($walker->getQuery()->hasHint('dummy-hint'));
+        self::assertSame('dummy-value', $walker->getQuery()->getHint('dummy-hint'));
+        self::assertInstanceOf(Query\Parameter::class, $param = $walker->getQuery()->getParameter('dummy-param'));
+        self::assertSame(123, $param->getValue());
+        self::assertFalse($walker->getQuery()->isCacheable());
```

Comment:
The cloning is an essential part of your PR. Please check that both queries and their Parameter objects  aren't the same instances.

Reply:
Code for cloning is copied from `\Doctrine\ORM\Tools\Pagination\Paginator::cloneQuery()`.

Added check for `Query` instances, but can't do it for `Parameter` instances, cause the way the code is, it just clones the `ArrayCollection` that contains the `Parameter` instances.

For my use case (bug), cloning the parameters or setting the cloned query as non-cacheable is really not needed. What I really need is cloning the query hints along with the query.

_I don't have the knowledge on this library to decide on this, just heard about SQLWalkers 2 days ago 😅_

Reply:
Then comparing both query instances and their `getParameters()` results would be sufficient. Just saw you already did compare the query instances, so you just need to add the second clones check.

``` php
self::assertNotSame($query->getParameters(), $walker->getQuery()->getParameters());
```

Reply:
Done!